river/tree/_utils/gradient_hessian.py

```diff
@@ -0,0 +1,130 @@
+import copy
+
+from river import stats
+
+
+class GradHess:
+    """ The most basic inner structure of the Streaming Gradient Trees. """
+
+    __slots__ = ["gradient", "hessian"]
+
+    def __init__(self, gradient: float = 0.0, hessian: float = 0.0, *, grad_hess=None):
+        if grad_hess:
+            self.gradient = grad_hess.gradient
+            self.hessian = grad_hess.hessian
+        else:
+            self.gradient = gradient
+            self.hessian = hessian
+
+    def __iadd__(self, other):
+        self.gradient += other.gradient
+        self.hessian += other.hessian
+
+        return self
+
+    def __isub__(self, other):
+        self.gradient -= other.gradient
+        self.hessian -= other.hessian
+
+        return self
+
+    def __add__(self, other):
+        new = copy.deepcopy(self)
+        new += other
+        return new
+
+    def __sub__(self, other):
+        new = copy.deepcopy(self)
+        new -= other
+        return new
+
+
+class GradHessStats:
+    """ Class used to monitor and update the gradient/hessian information in Streaming Gradient
+    Trees.
+
+    Represents the aggregated gradient/hessian data in a node (global node statistics), category,
+    or numerical feature's discretized bin.
+    """
+
+    def __init__(self):
+        self.x_m = stats.Mean()
```

Comment:
What about `x_mean`?

Reply:
Agreed! Changed in commit [940122d](https://github.com/online-ml/river/pull/458/commits/940122dfb81b6e62d7f43a2e51cfb113b90d899d).